subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputIntegrationTest.groovy

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.reflect.problems.ValidationProblemId
+import org.gradle.internal.reflect.validation.ValidationTestFor
+import spock.lang.Issue
+
+class TaskInputIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
❓ Why do we need this new type, and why add a very similar test to the one in `ValidatePluginsIntegrationTest`?

Reply:
I was under the impression that we need one test for building a plugin (1, `ValidatePluginsIntegrationTest`) and one test for running a build that uses a task from a plugin (2, the new `TaskInputIntegrationTest`) - but I might be wrong?